tests/validation/tests/v3_api/test_istio.py

```diff
@@ -0,0 +1,319 @@
+            ISTIO_INGRESSGATEWAY_NODEPORT,
+        "gateways.istio-ingressgateway.ports[0].port": 80,
+        "gateways.istio-ingressgateway.ports[0].targetPort": 80,
+        "gateways.istio-ingressgateway.ports[0].name": "http2",
+    }
+    external_id = "catalog://?catalog=system-library" + \
+                  "&template=rancher-istio" + \
+                  "&version=" + version
+    print("creating istio catalog app")
+    app = p_client.create_app(
+        name="cluster-istio",
+        externalId=external_id,
+        targetNamespace=ns.name,
+        projectId=project.id,
+        answers=answers
+    )
+    print("Verify istio app installed condition")
+    wait_for_condition(
+        p_client, app, check_condition('Installed', 'True'), 120)
+
+    print("Verify istio app deployment condition")
+    wait_for_condition(
+        p_client, app, check_condition('Deployed', 'True'), 600)
+    return app
+
+
+def verify_admission_webhook():
+    has_admission_webhook = execute_kubectl_cmd(
+        'api-versions | grep admissionregistration', False)
+    if len(has_admission_webhook) == 0:
+        raise AssertionError(
+            "MutatingAdmissionWebhook and ValidatingAdmissionWebhook plugins "
+            "are not listed in the kube-apiserver --enable-admission-plugins")
+
+
+def add_istio_label_to_ns(c_client, ns):
+    labels = {
+        "istio-injection": "enabled"
+    }
+    ns = c_client.update_by_id_namespace(ns.id, labels=labels)
+    return ns
+
+
+def create_and_test_bookinfo_services(p_client, ns, timeout=DEFAULT_TIMEOUT):
+    book_info_file_path = ISTIO_PATH + '/bookinfo.yaml'
+    execute_kubectl_cmd('apply -f ' + book_info_file_path + ' -n '
+                        + ns.name, False)
+    result = execute_kubectl_cmd('get deployment -n ' + ns.name, True)
+
+    for deployment in result['items']:
+        wl = wait_for_wl_by_id_to_active(p_client, deployment, 60)
+        wl_pods = wait_for_pods_in_workload(p_client, wl, 1)
+        wait_for_pod_to_running(p_client, wl_pods[0])
+
+    rating_pod = execute_kubectl_cmd('get pod -l app=ratings -n' + ns.name)
+    assert len(rating_pod['items']) == 1
+
+    rating_pod_name = rating_pod['items'][0]['metadata']['name']
+    result = execute_kubectl_cmd(
+        'exec -it -n ' + ns.name + ' ' + rating_pod_name
+        + ' -c ratings -- curl productpage:9080/productpage'
+        + ' | grep -o "<title>.*</title>"', False)
+
+    start = time.time()
+    while result.rstrip() != ISTIO_BOOKINFO_QUERY_RESULT:
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out and failed to get bookinfo service ready")
+        time.sleep(.5)
+        result = execute_kubectl_cmd(
+            'exec -it -n ' + ns.name + ' ' + rating_pod_name
+            + ' -c ratings -- curl productpage:9080/productpage'
+            + ' | grep -o "<title>.*</title>"', False)
+        assert result.rstrip() == ISTIO_BOOKINFO_QUERY_RESULT
```

Comment:
I doubt this assertion makes the while loop runs at most one iteration, and actually the assertion can be removed.